### PR TITLE
XRDDEV-1894

### PR DESCRIPTION
**Pull request by jhyoty**

### CODE REVIEW

**Review by teemukin65 — CHANGES_REQUESTED** [review]

File: src/centralserver/admin-service/src/main/java/org/niis/xroad/centralserver/restapi/repository/MemberClassRepository.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import org.springframework.stereotype.Repository;
+
+import java.util.Optional;
+import java.util.Set;
+
+@Repository
+public interface MemberClassRepository extends CrudRepository<MemberClass, Long> {
+    Optional<MemberClass> findByCodeIgnoreCase(String code);
+
+    Set<MemberClassDto> findAllAsDtoBy();
```
[the remark this answers is not in the document]
Comment (jhyoty) [reply src/centralserver/admin-service/src/main/java/org/niis/xroad/centralserver/restapi/repository/MemberClassRepository.java]: True, but bare findBy() (which would be the minimal acceptable) would hardly be a good name here.

File: src/centralserver/admin-service/src/main/java/org/niis/xroad/centralserver/restapi/service/MemberClassService.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+    /**
+     * Add a new member class
+     * @param memberClassDto
+     * @return
+     */
+    public MemberClassDto add(final MemberClassDto memberClassDto) {
+        return toDto(repository.findByCodeIgnoreCase(memberClassDto.getCode())
+                .orElseGet(() -> repository.save(
+                        new MemberClass(memberClassDto.getCode(), memberClassDto.getDescription()))));
```
[the remark this answers is not in the document]
Comment (jhyoty) [reply src/centralserver/admin-service/src/main/java/org/niis/xroad/centralserver/restapi/service/MemberClassService.java]: Good catch. The old implementation always converts codes to upper case.

File: src/centralserver/admin-service/src/test/java/org/niis/xroad/centralserver/restapi/openapi/MemberClassControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    private MemberClassController memberClassController;
+
+    @Test
+    @WithMockUser(authorities = {"ADD_MEMBER_CLASS"})
+    public void testAddMemberClass() {
+        final MemberClass mc = new MemberClass();
+        mc.setCode("FOO");
+        mc.setDescription("Description");
+        ResponseEntity<MemberClass> response = memberClassController.addMemberClass(mc);
+    }
```
[the remark this answers is not in the document]
Comment (jhyoty) [reply src/centralserver/admin-service/src/test/java/org/niis/xroad/centralserver/restapi/openapi/MemberClassControllerTest.java:51]: Assertions were added in a later commit.

File: src/centralserver/admin-ui/vue.config.js [hunk 4]
```diff
@@ -34,4 +34,13 @@ module.exports = {
       },
     },
   },
+  chainWebpack: config => {
+    config
+      .plugin('fork-ts-checker')
+      .tap(args => {
+        //args[0].memoryLimit = 4096;
+        //args[0].workers = 2;
+        return args
```
Comment (teemukin65) [inline src/centralserver/admin-ui/vue.config.js:43]: Is this some hint for future use cases or should this '.tap - ...' statement actually be removed for it's just commented out code?

File: src/common-db/src/main/java/ee/ria/xroad/common/db/CustomPostgreSQLDialect.java [hunk 5]
```diff
@@ -30,5 +30,6 @@
 /**
  * @deprecated To be removed
  */
+@Deprecated
 public class CustomPostgreSQLDialect extends PostgreSQL9Dialect {
```
Comment (teemukin65) [inline src/common-db/src/main/java/ee/ria/xroad/common/db/CustomPostgreSQLDialect.java:34]: Could this whole file be removed, if it is deprecated?   
At least it looks empty.

File: src/common-rest-api/src/main/java/org/niis/xroad/restapi/exceptions/ApplicationExceptionHandler.java [hunk 6]
```diff
@@ -83,7 +83,11 @@
                 RestApiAuditEvent.API_KEY_AUTHENTICATION, RestApiAuditEvent.AUTH_CREDENTIALS_DISCOVERY)) {
             auditEventLoggingFacade.auditLogFail(RestApiAuditEvent.UNSPECIFIED_AUTHENTICATION, e);
         }
-        log.error(EXCEPTION_CAUGHT, e);
+        if (log.isDebugEnabled()) {
+            log.error(EXCEPTION_CAUGHT, e);
+        } else {
+            log.error("Authentication failure: {}", e.getMessage());
+        }
         return exceptionTranslator.toResponseEntity(e, HttpStatus.UNAUTHORIZED);
```
Comment (teemukin65) [inline src/common-rest-api/src/main/java/org/niis/xroad/restapi/exceptions/ApplicationExceptionHandler.java:91]: Good, this makes cleaner logs in normal operation!

**Review by teemukin65 — CHANGES_REQUESTED** [review]
Biggish pile to review in one go. But some findings, Some just notions. And I really learned some tips & tricks on how these could be done!

File: src/centralserver/admin-ui/src/views/GlobalResources/GlobalGroup/GroupMembersFilterDialog.vue [hunk 7]
```diff
@@ -1,3 +1,28 @@
...
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+   THE SOFTWARE.
+ -->
 <!--
```
[the remark this answers is not in the document]
Comment (jhyoty) [reply src/centralserver/admin-ui/src/views/GlobalResources/GlobalGroup/GroupMembersFilterDialog.vue:26]: Hmm, perhaps the license-add npm task does not work as intended.

File: src/common-db/src/main/java/ee/ria/xroad/common/db/CustomPostgreSQLDialect.java (hunk 5 above)
Comment (jhyoty) [reply src/common-db/src/main/java/ee/ria/xroad/common/db/CustomPostgreSQLDialect.java:34]: It is possibly referenced in some scripts, not in scope of this ticket. Annotation added to silence a compiler warning.

**Review by teemukin65 — APPROVED** [review]

**Pull request merged**